shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -1269,12 +1514,60 @@ - (BOOL)hasText {
 }
 
 - (void)insertText:(NSString*)text {
+  NSMutableArray<NSArray<NSNumber*>*>* copiedRects =
+      [[NSMutableArray alloc] initWithCapacity:[_selectionRects count]];
+  NSAssert([_selectedTextRange.start isKindOfClass:[FlutterTextPosition class]],
+           @"Expected a FlutterTextPosition for position (got %@).",
+           [_selectedTextRange.start class]);
+  NSUInteger insertPosition = ((FlutterTextPosition*)_selectedTextRange.start).index - 1;
+  NSUInteger insertIndex = 0;
+  for (NSUInteger i = 0; i < [_selectionRects count]; i++) {
+    NSUInteger rectPosition = [_selectionRects[i][4] unsignedIntegerValue];
+    if (rectPosition == insertPosition) {
+      insertIndex = i;
+      for (NSUInteger j = 0; j <= text.length; j++) {
+        [copiedRects addObject:@[
+          _selectionRects[i][0],
+          _selectionRects[i][1],
+          _selectionRects[i][2],
+          _selectionRects[i][3],
+          [NSNumber numberWithInt:rectPosition + j],
+        ]];
+      }
+    } else {
+      if (rectPosition > insertPosition) {
+        rectPosition = rectPosition + text.length;
+      }
+      [copiedRects addObject:@[
+        _selectionRects[i][0],
+        _selectionRects[i][1],
+        _selectionRects[i][2],
+        _selectionRects[i][3],
+        [NSNumber numberWithInt:rectPosition],
```

Comment:
Since we are already extending the `UITextSelectionRect` class as `FlutterTextSelectionRect`, I went ahead and added a `position` prop there and refactored the code accorditly